node/blockAPI/shardBlock_test.go

```diff
@@ -0,0 +1,270 @@
+	assert.Nil(t, blk)
+	assert.Error(t, err)
+}
+
+func TestShardAPIBlockProcessor_GetBlockByHashFromNormalNode(t *testing.T) {
+	t.Parallel()
+
+	nonce := uint64(1)
+	round := uint64(2)
+	epoch := uint32(1)
+	shardID := uint32(3)
+	miniblockHeader := []byte("miniBlockHash")
+	headerHash := []byte("d08089f2ab739520598fd7aeed08c427460fe94f286383047f3f61951afc4e00")
+
+	storerMock := mock.NewStorerMock()
+	uint64Converter := mock.NewNonceHashConverterMock()
+
+	shardAPIBlockProcessor := createMockShardAPIProcessor(
+		shardID,
+		headerHash,
+		storerMock,
+		false,
+		true,
+	)
+
+	header := &block.Header{
+		Nonce:   nonce,
+		Round:   round,
+		ShardID: shardID,
+		Epoch:   epoch,
+		MiniBlockHeaders: []block.MiniBlockHeader{
+			{Hash: miniblockHeader},
+		},
+		AccumulatedFees: big.NewInt(0),
+		DeveloperFees:   big.NewInt(0),
+	}
+	headerBytes, _ := json.Marshal(header)
+	_ = storerMock.Put(headerHash, headerBytes)
+
+	nonceBytes := uint64Converter.ToByteSlice(nonce)
+	_ = storerMock.Put(nonceBytes, headerHash)
+
+	expectedBlock := &api.Block{
+		Nonce: nonce,
+		Round: round,
+		Shard: shardID,
+		Epoch: epoch,
+		Hash:  hex.EncodeToString(headerHash),
+		MiniBlocks: []*api.MiniBlock{
+			{
+				Hash: hex.EncodeToString(miniblockHeader),
+				Type: block.TxBlock.String(),
+			},
+		},
+		AccumulatedFees: "0",
+		DeveloperFees:   "0",
+		Status:          BlockStatusOnChain,
+	}
+
+	blk, err := shardAPIBlockProcessor.GetBlockByHash(headerHash, false)
+	assert.Nil(t, err)
+	assert.Equal(t, expectedBlock, blk)
+}
+
+func TestShardAPIBlockProcessor_GetBlockByNonceFromHistoryNode(t *testing.T) {
+	t.Parallel()
+
+	nonce := uint64(1)
+	round := uint64(2)
+	epoch := uint32(1)
+	shardID := uint32(3)
+	miniblockHeader := []byte("miniBlockHash")
+	headerHash := []byte("d08089f2ab739520598fd7aeed08c427460fe94f286383047f3f61951afc4e00")
+
+	storerMock := mock.NewStorerMock()
+
+	shardAPIBlockProcessor := createMockShardAPIProcessor(
+		shardID,
+		headerHash,
+		storerMock,
+		true,
+		false,
+	)
+
+	header := &block.Header{
+		Nonce:   nonce,
+		Round:   round,
+		ShardID: shardID,
+		Epoch:   epoch,
+		MiniBlockHeaders: []block.MiniBlockHeader{
+			{Hash: miniblockHeader},
+		},
+		AccumulatedFees: big.NewInt(100),
+		DeveloperFees:   big.NewInt(50),
+	}
+	headerBytes, _ := json.Marshal(header)
+	_ = storerMock.Put(headerHash, headerBytes)
+
+	expectedBlock := &api.Block{
+		Nonce: nonce,
+		Round: round,
+		Shard: shardID,
+		Epoch: epoch,
+		Hash:  hex.EncodeToString(headerHash),
+		MiniBlocks: []*api.MiniBlock{
+			{
+				Hash: hex.EncodeToString(miniblockHeader),
+				Type: block.TxBlock.String(),
+			},
+		},
+		AccumulatedFees: "100",
+		DeveloperFees:   "50",
+		Status:          BlockStatusOnChain,
+	}
+
+	blk, err := shardAPIBlockProcessor.GetBlockByNonce(1, false)
+	assert.Nil(t, err)
+	assert.Equal(t, expectedBlock, blk)
+}
+
+func TestShardAPIBlockProcessor_GetBlockByHashFromHistoryNodeStatusReverted(t *testing.T) {
```

Comment:
also here add a test with a block with transactions

Reply:
same, will be added after refactor.